physx/src/scene.rs

```diff
@@ -630,14 +633,135 @@ pub trait Scene: Class<physx_sys::PxScene> + UserData {
+    /// For simple usage, passing in [`PxRaycastBuffer`] will suffice.
+    /// Users can also provide a custom implementation of [`RaycastCallback`] in cases where allocating a single buffer to store
+    /// hits is prohibitively expensive.
+    /// Returns true if the ray hit an actor.
+    /// See the [PhysX documentation on `raycast`][src] for more details.
+    ///
+    /// [src]: (https://gameworksdocs.nvidia.com/PhysX/4.1/documentation/physxapi/files/classPxScene.html#a7d7dcd877cee092f8b57c67d79982b50)
+    #[allow(clippy::too_many_arguments)]
+    fn raycast(
+        &self,
+        origin: &PxVec3,
+        unit_dir: &PxVec3,
+        distance: f32,
+        hit_callback: &mut impl RaycastCallback,
+        hit_flags: HitFlags,
+        filter_data: &PxQueryFilterData,
+        filter_callback: Option<&mut PxQueryFilterCallback>,
+        cache: Option<&PxQueryCache>,
+    ) -> bool {
+        let filter_callback = filter_callback
+            .map_or(std::ptr::null_mut::<PxQueryFilterCallback>(), |v| {
+                v as *mut _
+            });
+        let cache = cache.map_or(std::ptr::null::<PxQueryCache>(), |v| v as *const _);
+
+        unsafe {
+            let hit_callback = hit_callback.into_px();
+
+            PxScene_raycast(
+                self.as_ptr(),
+                origin.as_ptr(),
+                unit_dir.as_ptr(),
+                distance,
+                hit_callback,
+                hit_flags.into_px(),
+                &filter_data.obj,
+                filter_callback,
+                cache,
+            )
+        }
+    }
+
+    /// Sweeps a shape across the scene, reporting any hit objects via the [`SweepCallback`] trait passed into `hit_callback`.
+    /// For simple usage, passing in [`PxSweepBuffer`] will suffice.
+    /// Users can also provide a custom implementation of [`SweepCallback`] in cases where allocating a single buffer to store
+    /// hits is prohibitively expensive.
+    /// Returns true if the sweep hit an actor.
+    /// See the [PhysX documentation on `sweep`][src] for more details.
+    ///
+    /// [src]: (https://gameworksdocs.nvidia.com/PhysX/4.1/documentation/physxapi/files/classPxScene.html#a9b07b2a98e64105a06e97ffaeba2a63d)
+    #[allow(clippy::too_many_arguments)]
+    fn sweep(
+        &self,
+        geometry: &impl Geometry,
+        pose: &PxTransform,
+        direction: &PxVec3,
+        distance: f32,
+        hit_callback: &mut impl SweepCallback,
+        hit_flags: HitFlags,
+        filter_data: &PxQueryFilterData,
+        filter_callback: Option<&mut PxQueryFilterCallback>,
+        cache: Option<&PxQueryCache>,
+        inflation: f32,
+    ) -> bool {
+        let filter_callback = filter_callback
+            .map_or(std::ptr::null_mut::<PxQueryFilterCallback>(), |v| {
+                v as *mut _
+            });
+        let cache = cache.map_or(std::ptr::null::<PxQueryCache>(), |v| v as *const _);
```

Comment:
I personally find `.map(...).unwrap_or(...)` clearer. Not a required change, but it's an idiom I see a lot more in the wild as opposed to `map_or`.